File: metodo_shell.py
#metodo de ordenamiento shell sort
def shell_sort(arr):
    n = len(arr)
    gap = n // 2  # Inicializa el gap

    # Reduce el gap hasta que sea 0
    while gap > 0:
        for i in range(gap, n):
            temp = arr[i]
            j = i
            # Mueve los elementos del arreglo que son mayores que temp
            while j >= gap and arr[j - gap] > temp:
                arr[j] = arr[j - gap]
                j -= gap
            arr[j] = temp
        gap //= 2  # Reduce el gap

    return arr

# Ejemplo de uso
numeros = [12, 34, 54, 2, 3]
print("Arreglo original:", numeros)
sorted_numeros = shell_sort(numeros)
print("Arreglo ordenado:", sorted_numeros)

File: metodo_shell.cs
using System;
class metodo_shell
{
    // metodo de ordenamiento shell sort
    public static void Shell(int[] arreglo)
    {
        int salto, aux, i;
        bool cambios;
        for (salto = arreglo.Length / 2; salto != 0; salto /= 2)
        {
            cambios = true;
            while (cambios)
            {
                cambios = false;
                for (i = salto; i < arreglo.Length; i++)
                {
                    if (arreglo[i - salto] > arreglo[i])
                    {
                        aux = arreglo[i];
                        arreglo[i] = arreglo[i - salto];
                        arreglo[i - salto] = aux;
                        cambios = true;
                    }
                }
            }
        }
    }

    static void Main()
    {
        int[] arreglo = { 23, 12, 1, 8, 34, 54, 2, 3 };
        Console.WriteLine("Arreglo original:");
        Console.WriteLine(string.Join(", ", arreglo));
        Shell(arreglo);
        Console.WriteLine("Arreglo ordenado:");
        Console.WriteLine(string.Join(", ", arreglo));
    }
}